Src/DependencyCollector/Shared/TelemetryDiagnosticSourceListener.cs

```diff
@@ -209,6 +209,17 @@ protected override HashSet<string> GetListenerContext(DiagnosticListener diagnos
             return includedActivities;
         }
 
+        protected override HandleDiagnosticsEvent<HashSet<string>> GetEventHandler(DiagnosticListener diagnosticListener)
+        {
+            HandleDiagnosticsEvent<HashSet<string>> eventHandler;
+            if (this.customEventHandlers.TryGetValue(diagnosticListener.Name, out eventHandler))
+            {
+                return eventHandler;
```

Comment:
So this skips the ```IsActivityIncluded()``` check which allows the user to control which individual activities to track

Reply:
not the user, but another handler. I'm fine with adding one more check for IsEventEnabled here  and removing it from handler completely. BTW, it will remove the need to TContext in handler

Reply:
This check is done in IsEventEnabled, it's also done within a handler.
The point is handler is not requested for the source that is disabled

IsActivityEnabled is also called in this IsEventEnabled, i.e. users may filter out (as tests demonstate)

The only problem is multiple listeners when event may be fired even if your listener returned false in IsEnabled. That's a problem in ServiceBus listener indeed. Will fix

Reply:
I have separated activity filtering (based on the config) from the particular event handler.
I.e. http handler may have more sophisticated logic to filter events than the EventHub listener.

This was filtering by activity name is part of the TelemetryDiagnosticSourceListener and done before any handler. It allows to remove this logic from the handler, which then could only parse specific event.

Let me know what do you think.

I also expect that in future there will be a need to process somehow start events (e.g. Inject app in when sending data and supporting multi ikey scenarios) - it's not possible now and I don't want to ix it until strictly necessary.

Reply:
Looks good - thanks!